src/main/java/mil/dds/anet/database/PositionDao.java

```diff
@@ -461,4 +461,57 @@ public static String generateCurrentPositionFilter(String personJoinColumn,
             + " WHERE pp.\"positionUuid\" = :%3$s AND maxPp.\"createdAt\" IS NULL ",
         personJoinColumn, dateFilterColumn, placeholderName);
   }
+
+  @InTransaction
+  public int cleanLoserPositionCode(String loserUuid) {
+    return getDbHandle().createUpdate(
+        "/* deleteLoserPositionCodeBeforeMerge */ UPDATE \"positions\" set \"code\" = NULL "
+            + "WHERE \"uuid\" = :loserUuid")
+        .bind("loserUuid", loserUuid).execute();
+  }
+
+  @InTransaction
+  public int mergePositions(Position existingPos, Position mergedPosition) {
+    getDbHandle().createUpdate(
+        "/* updatePeoplePositionsAfterMergePosition */ UPDATE \"peoplePositions\" set \"positionUuid\" = :mergedPositionUuid "
+            + "WHERE \"positionUuid\" = :existingPosUuid")
+        .bind("mergedPositionUuid", mergedPosition.getUuid())
+        .bind("existingPosUuid", existingPos.getUuid()).execute();
+
+    getDbHandle().execute(
+        "DELETE FROM \"positionRelationships\" WHERE \"positionUuid_a\" = ? OR \"positionUuid_b\"= ?",
+        existingPos.getUuid(), existingPos.getUuid());
+
+    // Update notes of existing position's with merged position info
+    getDbHandle().createUpdate(
+        "/* updateNoteRelatedObjectsAfterMergePosition */ UPDATE \"noteRelatedObjects\" set \"relatedObjectUuid\" = :mergedPositionUuid "
+            + "WHERE \"relatedObjectUuid\" = :existingPosUuid")
+        .bind("mergedPositionUuid", mergedPosition.getUuid())
+        .bind("existingPosUuid", existingPos.getUuid()).execute();
```

Comment:
This will fail if there's a note pointing to *both* positions. See https://github.com/NCI-Agency/anet/blob/GH-2995-improve-and-extend-merge-for-positions-object/src/main/java/mil/dds/anet/database/PersonDao.java#L407-L418 in `PersonDao::mergePeople` for a solution.